server/src/main/java/org/elasticsearch/script/DoubleFieldScript.java

```diff
@@ -21,6 +22,25 @@
     @SuppressWarnings("unused")
     public static final String[] PARAMETERS = {};
 
+    public static DoubleFieldScript.Factory factory(Consumer<DoubleFieldScript> executor) {
```

Comment:
This method, in combination with making `emit` public, makes writing tests so much easier,

Reply:
Would it help similarly if `DoubleFieldScript` implements an interface and tests didn't use the `AbstractFieldScript` infrastructure at all? Its kind of designed around incantations that work well in painless which isn't *really* a concern that the mapping infrastructure needs to have. Maybe it'd be easier to reason about with an interface and the tests just implementing that directly? I dunno.

Reply:
I looked at that, but this turns out to be simpler - the interface ends up basically duplicating all of the logic except construction.

Reply:
do we end up executing the script in these tests? I was wondering if instead of taking a consumer of the field script, this method should take a supplier of field script, then the test would subclass the script class directly... maybe you have already tried it though. Also maybe this could be one method in the base mapper script test class with a generic type?

Reply:
> do we end up executing the script in these tests?

Yes - see for example `testMultipleValues` in `LongScriptMapperTests`

The reason I've done it this way is that the ceremony is all around construction, and returning a subclass doesn't help there because you still need to pass all the parameters to the super constructor.  And you can't construct a generic class without knowing the type, so it can't be a method on the base test class.

Reply:
makes sense, can we then move these new static  methods to their corresponding test classes, assuming they are only used there?